src/augtool.c

```diff
@@ -481,7 +534,14 @@ static int main_loop(void) {
 
     while(1) {
         if (get_line) {
```

Comment:
following the logic, and the variables' malloc/free/assignment is really hard in this loop :(
how hard would it be to put all of it in very small (inlined?) functions?